### PR TITLE
Tweac local

### PR DESCRIPTION
**Pull request by HaritzPuerto**
# What does this PR do?
The backend of TWEAC is working now! However, I could only try it with a mix of my local and the server... (i.e., with mock responses). Full testing is needed in the server

Right now, the model service can only return 1 prediction, so tweac can only return 1 dataset prediction.

The TWEAC Skill is only returning calling to 1 Skill, instead of top 3 because the QueryResponse object is only for 1 Skill. Let's check first that this is working and then, let's allow calling more than 1 Skill

### CODE REVIEW

File: skill-manager/api_server.http [hunk 1]
```diff
@@ -0,0 +1,136 @@
+@hostname = square.ukp-lab.de
```
Comment (timbmg) [inline skill-manager/api_server.http:1]: is this file really required? If the only thing changes is the hostname it would be better to just edit it in the file to the host that you are targeting. When having two copies of the same file they will get out of sync for sure 😅

File: skill-manager/skill_manager/routers/skill.py [hunk 2]
```diff
@@ -72,6 +72,40 @@ async def get_skills(request: Request):
...
+        payload = await get_payload_from_token(request)
+        user_id = payload["username"]
+        # make a mongo query to get all the skills that the user has access to and are published and trained on the given dataset
+        mongo_query = {
+            "$and": [
+                {"data_sets": dataset},
+                {
+                    "$or": [
+                        mongo_query,
+                        # {"user_id": user_id}, # this user id is not the one that logged in
```
Comment (timbmg) [inline skill-manager/skill_manager/routers/skill.py:92]: but it should be the user id of the user that actually triggered the request, right? Could we make the user_id another parameter of this endpoint? Ie the meta-skill will call this endpoit with the token (that has the meta-skill as user_id), however, the meta-skill can know who called it and pass the user_id to this endpoint too

File: skills/local/skill.py [hunk 3]
```diff
@@ -17,12 +17,64 @@
...
+    query = request.query
+
+    # Call Model API
+    model_request = {
+        "input": [query],
+    }
+
+    logger.debug("Request for model api:{}".format(model_request))
+
+    # model_response = await square_model_client(
```
Comment (timbmg) [inline skills/local/skill.py:32]: remove, it no longer needed

File: skills/local/skill.py [hunk 4]
```diff
@@ -17,12 +17,64 @@
...
+    #     model_name=request.skill_args["base_model"],
+    #     pipeline="sequence-classification",
+    #     model_request=model_request,
+    # )
```
Comment (timbmg) [inline skills/local/skill.py:36]: remove, if no longer needed

File: skills/local/skill.py [hunk 5]
```diff
@@ -17,12 +17,64 @@
 
 async def predict(request: QueryRequest) -> QueryOutput:
 
+    """
```
Comment (timbmg) [inline skills/local/skill.py:20]: this file was originally intended to have the most simple, hardcoded skill for local setups. Now it seems a bit messy. Are these changes needed? It seems this was more for your development?

File: skills/tweac/skill.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        "input": [request.query],
+    }
+    logger.debug("Request for model api:{}".format(model_request))
+
+    model_response = await square_model_client(
+        model_name=request.skill_args["base_model"],
+        pipeline="sequence-classification",
+        model_request=model_request,
+    )
+    # model_response = {'model_outputs': {'logits': None},
```
Comment (timbmg) [inline skills/tweac/skill.py:44]: please remove commented code

File: skill-manager/api_server.http (hunk 1 above)
Comment (HaritzPuerto) [reply skill-manager/api_server.http:1]: strictly speaking, it's not necessary, but it is very beneficial for future developers. I added a lot of possible api calls to the skill manager or related. The change is not the hostname, but all other api calls.

File: skills/local/skill.py (hunk 5 above)
Comment (HaritzPuerto) [reply skills/local/skill.py]: sorry, that was an accident, that was some testing for tweac. Already fixed

File: skills/tweac/skill.py (hunk 6 above)
Comment (HaritzPuerto) [reply skills/tweac/skill.py]: done

**Comment by HaritzPuerto** [comment]
image

TWEAC is working on extractive Skills! However, we don't show the Skill that was used to solve the question

File: skill-manager/api_server.http (hunk 1 above)
Comment (timbmg) [reply skill-manager/api_server.http:1]: Ok, i might need to have a closer look. But why not just extend api.http?

Comment (HaritzPuerto) [reply skill-manager/api_server.http]: oh! you mean that! sorry you are right. I think this file was made as a backup and I guess accidentally I included in the commit

Comment (HaritzPuerto) [reply skill-manager/api_server.http]: I have deleted it
